### PR TITLE
modul1

### PR DESCRIPTION
**Pull request by pilinoga**
add first module to review

### CODE REVIEW

File: module1/build tools/core/src/main/java/com/epam/core/Utils.java [hunk 1]
```diff
@@ -0,0 +1,16 @@
+package com.epam.core;
+
+import com.epam.task1.StringUtils;
+import java.util.Arrays;
+
+public class Utils {
+
+    public static void main(String[] args) {
```
Comment (senator2202) [inline module1/build tools/core/src/main/java/com/epam/core/Utils.java]: Why do you need main method inside this class?

File: module1/build tools/multi-project/src/main/java/Main.java [hunk 2]
```diff
@@ -0,0 +1,6 @@
+
+public class Main {
```
Comment (senator2202) [inline module1/build tools/multi-project/src/main/java/Main.java]: I don't see multiple modules inside multi-project. According to the task, you need to create **one** multi-project with **two** sub-projects. Now I can see 3 projects in diffrent directories

File: module1/build tools/multi-project/src/main/java/Main.java [hunk 3]
```diff
@@ -0,0 +1,6 @@
...
+
+    public static void main(String[] args) {
```
Comment (senator2202) [inline module1/build tools/multi-project/src/main/java/Main.java]: Why do you need main method is this class?

File: module1/build tools/api/build.gradle [hunk 4]
```diff
@@ -0,0 +1,15 @@
...
+group 'com.epam'
+version '1.0'
+
+repositories {
+    mavenCentral()
+}
+
+dependencies {
+    testCompile group: 'junit', name: 'junit', version: '4.12'
+    compile files('C:\\Users\\kirill\\Desktop\\lab\\module1\\build tools\\jar\\core.jar')
```
Comment (senator2202) [inline module1/build tools/api/build.gradle]: According to the task, you need only one prepared jar: "utils-1.3.5.jar". You need to define dependency from core-module, not from core.jar

File: module1/build tools/multi-project/build.gradle [hunk 5]
```diff
@@ -0,0 +1,16 @@
...
+group 'com.epam'
+version '1.0'
+
+repositories {
+    mavenCentral()
+}
+
+dependencies {
+    testCompile group: 'junit', name: 'junit', version: '4.12'
+    compile files('C:\\Users\\kirill\\Desktop\\lab\\module1\\build tools\\jar\\core.jar')
```
Comment (senator2202) [inline module1/build tools/multi-project/build.gradle]: According to the task, multi-project consists of 2 sub-projects, but not depends on them.

File: module1/build tools/ustils/src/main/java/com/epam/task1/StringUtils.java [hunk 6]
```diff
@@ -0,0 +1,22 @@
...
+
+    public static boolean isPositiveNumber(String str) {
+        boolean result;
+        if(NumberUtils.isParsable(str)) {
+            Long fromString = NumberUtils.createLong(str);
+            result = fromString > 0;
+        }else{
+            result = false;
+        }
+        return result;
```
Comment (senator2202) [inline module1/build tools/ustils/src/main/java/com/epam/task1/StringUtils.java]: This 8 rows of code can be simplified to one.

**Review by senator2202 — CHANGES_REQUESTED** [review]

File: module1/build tools/core/src/main/java/com/epam/core/Utils.java [hunk 7]
```diff
@@ -0,0 +1,16 @@
...
+    }
+
+    public static boolean isAllPositiveNumbers(String... str) {
```
Comment (senator2202) [inline module1/build tools/core/src/main/java/com/epam/core/Utils.java:11]: Add documentation on util methods

File: module1/build tools/core/src/main/java/com/epam/core/Utils.java [hunk 8]
```diff
@@ -0,0 +1,16 @@
+package com.epam.core;
+
+import com.epam.task1.StringUtils;
+import java.util.Arrays;
+
+public class Utils {
+
+    public static void main(String[] args) {
+    }
+
```
Comment (senator2202) [inline module1/build tools/core/src/main/java/com/epam/core/Utils.java]: Add documentation on util methods

File: module1/build tools/core/src/main/java/com/epam/core/Utils.java [hunk 9]
```diff
@@ -0,0 +1,16 @@
...
+    public static boolean isAllPositiveNumbers(String... str) {
+        int before = str.length;
```
Comment (senator2202) [inline module1/build tools/core/src/main/java/com/epam/core/Utils.java]: What will happen if (str == null) ?

File: module1/build tools/core/src/main/java/com/epam/core/Utils.java [hunk 10]
```diff
@@ -0,0 +1,16 @@
...
+        int after = (int) Arrays.stream(str).filter(StringUtils::isPositiveNumber).count();
+        return (before != 0 && before == after);
```
Comment (senator2202) [inline module1/build tools/core/src/main/java/com/epam/core/Utils.java]: Rows 13-14 can be simplified if you use allMatch terminal operation

File: module1/build tools/ustils/src/main/java/com/epam/task1/StringUtils.java [hunk 11]
```diff
@@ -0,0 +1,22 @@
+package com.epam.task1;
+
+import org.apache.commons.lang3.math.NumberUtils;
+
+public class StringUtils {
+
+    public static void main(String[] args) {
```
Comment (senator2202) [inline module1/build tools/ustils/src/main/java/com/epam/task1/StringUtils.java]: Why do you need main method in this class?

File: module1/build tools/core/src/main/java/com/epam/core/Utils.java (hunk 1 above)
Comment (pilinoga) [reply module1/build tools/core/src/main/java/com/epam/core/Utils.java]: Main method is required to create jar file.

File: module1/build tools/ustils/src/main/java/com/epam/task1/StringUtils.java (hunk 11 above)
Comment (pilinoga) [reply module1/build tools/ustils/src/main/java/com/epam/task1/StringUtils.java]: Delete method.

File: module1/build tools/core/src/main/java/com/epam/core/Utils.java (hunk 10 above)
Comment (pilinoga) [reply module1/build tools/core/src/main/java/com/epam/core/Utils.java]: Simplified method .

File: module1/build tools/multi-project/src/main/java/Main.java (hunk 2 above)
Comment (pilinoga) [reply module1/build tools/multi-project/src/main/java/Main.java]: rebuild project, add two modules inside multi-project

File: module1/build tools/multi-project/src/main/java/Main.java (hunk 3 above)
Comment (pilinoga) [reply module1/build tools/multi-project/src/main/java/Main.java]: Reduild project, deleted class.

File: module1/build tools/core/src/main/java/com/epam/core/Utils.java (hunk 9 above)
Comment (pilinoga) [reply module1/build tools/core/src/main/java/com/epam/core/Utils.java]: Resolve case with null.

File: module1/build tools/ustils/src/main/java/com/epam/task1/StringUtils.java (hunk 6 above)
Comment (pilinoga) [reply module1/build tools/ustils/src/main/java/com/epam/task1/StringUtils.java]: Simplified method to one line.

File: module1/build tools/multi-project/build.gradle (hunk 5 above)
Comment (pilinoga) [reply module1/build tools/multi-project/build.gradle]: Added modules api and core to multi-project.

File: module1/build tools/api/build.gradle (hunk 4 above)
Comment (pilinoga) [reply module1/build tools/api/build.gradle]: Added dependency from core-module. Deleted dependency from core.jar.

File: module1/build tools/ustils/src/main/java/com/epam/task1/StringUtils.java (hunk 6 above)
Comment (senator2202) [reply module1/build tools/ustils/src/main/java/com/epam/task1/StringUtils.java]: @pilinoga can not see ustils module inside 'build tools' directory anymore. Only multi-project left

File: module1/build tools/multi-project/core/build.gradle [hunk 12]
```diff
@@ -0,0 +1,15 @@
...
+group 'com.epam'
+version '1.0'
+
+repositories {
+    mavenCentral()
+}
+
+dependencies {
+    testCompile group: 'junit', name: 'junit', version: '4.12'
+    compile files('C:\\Users\\kirill\\Desktop\\lab\\module1\\build tools\\jar\\utils-1.3.5.jar')
```
Comment (senator2202) [inline module1/build tools/multi-project/core/build.gradle]: @pilinoga,  If I decide to test your project on my local environment, my absolute path will be different. Use relative path to 'utils-1.3.5.jar' instead.

Comment (pilinoga) [reply module1/build tools/multi-project/core/build.gradle]: > @pilinoga, If I decide to test your project on my local environment, my absolute path will be different. Use relative path to 'utils-1.3.5.jar' instead.

Used relative path to utils-1.3.5.jar.

File: module1/build tools/ustils/src/main/java/com/epam/task1/StringUtils.java (hunk 6 above)
Comment (pilinoga) [reply module1/build tools/ustils/src/main/java/com/epam/task1/StringUtils.java]: @senator2202 returned utils module to remote repository.

**Review by senator2202 — APPROVED** [review]

**Comment by senator2202** [comment]
Add @Lavazz as a second reviewer

**Pull request merged**